File: 2025-08/2025-08-26/reverseParenthesis.js
function decode(s) {
  const re = /\([^\(\)]*\)/g;

  while(re.test(s)) {
    s = s.replace(re, replacer);
  }

  return s;
}

function replacer(match) {
  return match.slice(1, match.length - 1).split('').reverse().join('');
}

console.log(decode("(f(b(dc)e)a)"));
console.log(decode("((is?)(a(t d)h)e(n y( uo)r)aC)"));
console.log(decode("f(Ce(re))o((e(aC)m)d)p"));
console.log(decode("a()b"));

File: 2025-08/2025-08-26/reverse_parenthesis.py
import re

def decode(s):
    regex = re.compile(r'\([^\(\)]*\)')
    
    while regex.search(s):
        s = re.sub(regex, replacer, s)

    return s

def replacer(match):
    return match.group(0)[1:len(match.group(0)) - 1][::-1]

print(decode("(f(b(dc)e)a)"))
print(decode("((is?)(a(t d)h)e(n y( uo)r)aC)"))
print(decode("f(Ce(re))o((e(aC)m)d)p"))
print(decode("a(())b"))